scripts/functional-tests.sh

```diff
@@ -6,7 +6,24 @@ set -ex
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
-pip install .
+pip install -e .
```

Comment:
Why changing this line?

Reply:
In the scope of this test, there's no real reason to change it as it will `pip install` the version of the source at the time of running.

However; this becomes a problem when developing locally as the version of `ilab` now being used in the CLI will be a static version that's not reflective of any changes we make.

Reply:
Strong -1

Don't use `pip install -e` in tests. The option makes pip use a different approach inject a package into a virtual env. It does not actually install the package. Instead it drops a `.pth` file into site-packages and modifies `sys.path`. With `-e`, we are not testing what our users are deploying.

Reply:
@tiran That's a fair point, but how will we clean up the `pip install .`-ed package? I don't see anything in the script which actually cleans this up from the python environment after the tests finish. 

When users are developing locally and running these tests, it will be very confusing if their `ilab` CLI is suddenly fixed to a static version from the time of the last test build.

Reply:
The `pip install .` line should not be there in the first place. It's the job of tox to install and re-install the package. Tox monitors its virtual envs and takes care of installing dependencies or re-creating an outdated env. It will also rebuild and reinstall the current package every time you run a tox env.

If users want to re-use a virtual env as a development env, then they should run `pip install -e .`. Other options are `tox devenv` or changing `package = wheel` to `package = editable` in `tox.ini`.

Reply:
Is there anyway we can automate this? I feel very strongly that our tests should not be overwriting an editable installation of `ilab` into a static one during development.

Reply:
I made an issue for this here: https://github.com/instructlab/instructlab/issues/1641 

Removing the `pip install -e .` line here and closing this.